src/Actions/Selection.ts

```diff
@@ -138,6 +142,37 @@ export class ActionSelection {
         return Promise.resolve(true);
     }
 
+    @PrototypeReflect.metadata(SymbolMetadata.Action.isChange, true)
+    static byTextObject(args: {
+        textObject: TextObject
+    }): Thenable<boolean> {
+
+        const activeTextEditor = window.activeTextEditor;
+
+        if (! activeTextEditor) {
+            return Promise.resolve(false);
+        }
+
+        let ranges: Range[] = [];
+
+        activeTextEditor.selections.forEach(selection => {
+            const match = args.textObject.apply(selection.active);
+            if (match) {
+                ranges.push(match);
```

Comment:
We should expand current selections rather than replace them.
Something like this should be done:
```ts
ranges.push(match ? selection.union(match) : selection);
```